docs/documentation/server_admin/topics/sso-protocols/con-oidc-auth-flows.adoc

```diff
@@ -72,6 +72,27 @@ The _Client Credentials Grant_ creates a token based on the metadata and permiss
 
 See the <<_service_accounts,Service Accounts>> chapter for more information.
 
+[[_refresh_token_grant]]
+==== Refresh token grant
+
+By default, {project_name} return refresh tokens in the token responses from most of the flows. Some exceptions are implicit flow or client credentials grant described above.
+
+Refresh token is tight to the user session of SSO browser session and can be valid for the whole lifetime of the user session. But note that client should send refresh-token request at least once per
+specified intervals as otherwise the session can be considered "idle" and can expire. See the <<_timeouts,timeouts section>> for more information.
+
+{project_name} supports <<_offline-access,offline tokens>>, which can be used typically when client needs to use refresh token even if corresponding browser SSO session is already expired.
+
+[[_refresh_token_rotation]]
+===== Refresh token rotation
+
+It is possible to specify that refresh token is considered invalid once it is used. This means that client must always save the refresh token from the last refresh response because older refresh tokens,
```

Comment:
```suggestion
It is possible to specify that the refresh token is considered invalid once it is used. This means that client must always save the refresh token from the last refresh response because older refresh tokens,
```